Mathlib/Dynamics/PeriodicPts/Lemmas.lean

```diff
@@ -63,6 +65,53 @@ theorem Commute.minimalPeriod_of_comp_eq_mul_of_coprime {g : α → α} (h : Com
   · exact (isPeriodicPt_minimalPeriod _ _).const_mul _
   · exact (isPeriodicPt_minimalPeriod _ _).mul_const _
 
+section Fintype
+
+variable [Fintype α]
+
+open Fintype
+
+theorem minimalPeriod_le_card : minimalPeriod f x ≤ card α := by
+  rw [← periodicOrbit_length]
+  exact List.Nodup.length_le_card nodup_periodicOrbit
+
+theorem isPeriodicPt_factorial_card_of_mem_periodicPts (h : x ∈ periodicPts f) :
+    IsPeriodicPt f (card α)! x :=
+  isPeriodicPt_iff_minimalPeriod_dvd.mpr
+    (Nat.dvd_factorial (minimalPeriod_pos_of_mem_periodicPts h) minimalPeriod_le_card)
+
+theorem mem_periodicPts_iff_isPeriodicPt_factorial_card :
+    x ∈ periodicPts f ↔ IsPeriodicPt f (card α)! x where
+  mp := isPeriodicPt_factorial_card_of_mem_periodicPts
+  mpr h := minimalPeriod_pos_iff_mem_periodicPts.mp
+    (IsPeriodicPt.minimalPeriod_pos (Nat.factorial_pos _) h)
+
+open Finset in
+theorem mem_periodicPts_of_injective (h : Injective f) : x ∈ periodicPts f := by
+  have h₁ (a) (ha : a ∈ range (card α).succ) : f^[a] x ∈ Finset.univ := mem_univ _
+  rcases exists_ne_map_eq_of_card_lt_of_maps_to (by simp) h₁ with ⟨_, _, _, _, h₂, h₃⟩
+  rcases lt_or_gt_of_ne h₂ with h₂ | h₂
+  · exact mk_mem_periodicPts (by omega) (iterate_cancel h h₃.symm)
+  · exact mk_mem_periodicPts (by omega) (iterate_cancel h h₃)
+
+theorem injective_iff_forall_mem_periodicPts : Injective f ↔ periodicPts f = univ := by
```

Comment:
These two should assume `Finite` rather than `Fintype` since they don't mention `card` ...